pkg/gardenlet/controller/gardenlet/add.go

```diff
@@ -54,7 +54,7 @@ func (r *Reconciler) AddToManager(
 	}
 	if r.HelmRegistry == nil {
 		var err error
-		r.HelmRegistry, err = oci.NewHelmRegistry()
+		r.HelmRegistry, err = oci.NewHelmRegistry(r.GardenClient)
```

Comment:
We need a special handling for `gardenlet` here since it doesn't have the permission to access secrets in the `garden` namespace. See https://github.com/gardener/gardener/blob/master/docs/concepts/controller-manager.md#main-reconciler-2 for more information.

Reply:
Addressed by following process:
The pull secret to be used by the `gardenlet` must be created in the `garden` namespace of the garden cluster with the label `gardener.cloud/role=helm-pull-secret`. It is then replicated into the seed namespace by the `Main` reconciler. The `HelmRegistry` fetches the secret from there as the namespace is configured via the context variable.